staxing/assignment.py

```diff
@@ -265,8 +265,7 @@ def adjust_date_picker(self, driver, target, new_date):
         month, year = current.text.split(' ')
         month = months[month]
         year = int(year)
-
-        while year <= new_date.year and month < new_date.month:
+        while year <= new_date.year:
```

Comment:
So here's the reason the calendar is advancing almost a year instead of just a month or three. The year loop should terminate once the years match:
`while year < new_date.year:`

Reply:
Yeah I actually changed it already and fixed test cases 301-306. But I want to make another pull request when I fix 307 as well.